avocado/plugins/run.py

```diff
@@ -99,6 +99,12 @@ def configure(self, parser):
                             "system information (hardware details, profilers, "
                             "etc.). Current:  %(default)s")
 
+        parser.add_argument("--suite-order", default="variants-per-test",
+                            choices=("tests-per-variant",
+                                     "variants-per-test"),
+                            help="How to iterate through test suite and "
```

Comment:
OK, `--execution-order=tests-per-variant` and `--execution-order=variants-per-test` it is!